### ISSUE TITLE
Add "test microphone" button - audio preview

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#223]
A microphone loopback to compliment the test video button.

**Comment by zetok** [comment]
From #1202:

# 

There should be a mic feedback feature\* in the Audio/Video settings to make sure audio input/output work.

\* think Mumble/Steam

**Comment by antis81** [comment]
Though not completely finished, we have a mic feedback widget now -> closing.

**Comment by Zer0-One** [comment]
@antis81 the feedback widget only tells you about volume. it doesn't tell you anything about the quality of your audio, which is basically the whole point of having something like a "test microphone" button.

**Comment by tux3** [comment]
I guess we have EchoBot too, which is a bit like Skype's echo service.

**Comment by Zer0-One** [comment]
@tux3 unless you start creating profiles with echobot added as a friend by default (which doesn't sound like a good idea) then i think qTox should have its own facilities for testing audio.

**Comment by tux3** [comment]
Fair enough!

**Comment by Diadlo** [comment]
May be combine the implementation of point "add someone as a friend" from 
https://github.com/tux3/qTox/issues/2657
and this issue (add echo bot like first friend)?

**Comment by rodolpheh** [comment]
I would love to see such a feature. For now, when I have trouble communicating with a friend, I can't know for sure if the problem is located in qtox or not.

**Comment by macandchief** [comment]
I have a strange issue with the microphone. OS: BunsenLabs (i.e. Debian Jessie), qt: 5.3.2, Pulseaudio.

If I open the mixer and go to recordings, when qTox is on, nothing in the recordings tab, mic does not work. 

If I go to the audio/video tab in the qTox-settings and set the capture device to "Built-in Audio Analog Stereo" (instead of "Monitor of..."), microphone starts working (it's also shown in the mixer tab). However if I go back to the contact page, it stops working (nothing in the mixer recordings tab).

So basically it works, but not properly.

**Comment by zetok** [comment]
@macandchief your issue is #3538.

**Comment by macandchief** [comment]
Indeed, I see. Great to read that it'll be fixed in 1.6, thank you.
